code/controllers/subsystem/explosion.dm

```diff
@@ -153,12 +153,12 @@ SUBSYSTEM_DEF(explosions)
 		A.color = null
 		A.maptext = ""
 
-/proc/dyn_explosion(turf/epicenter, power, flash_range, adminlog = TRUE, ignorecap = TRUE, flame_range = 0, silent = FALSE, smoke = TRUE, list/explosion_multiplier = list(1, 1, 1, 1))
+/proc/dyn_explosion(turf/epicenter, power, flash_range, adminlog = TRUE, ignorecap = TRUE, flame_range = 0, silent = FALSE, type = /datum/effect_system/explosion/smoke, list/explosion_multiplier = list(1, 1, 1, 1))
```

Comment:
`type` is a keyword in byond, use a different variable name

Reply:
Alright it seems like you removed smoke = FALSE but didnt change the type to the explosion without smoke for a lot of explosions

Reply:
I changed the keyword to explosion_type. Also, dyn_explosion set smoke to true as default, and the base explosion proc set smoke to false as default. I updated the original proc and all uses of dyn_explosion() and explosion() to the new format, so this should be correct.